### PR TITLE
Add more info on troubleshooting the elastic stack

### PR DESCRIPTION
**Pull request by moskyb**
**This PR:** Adds a bunch of info around troubleshooting the elastic stack - notes on what logs get sent where, and how to get into an instance if it's misbehaving.

My wording probably isn't awesome here, feel free to edit as necessary :)

### CODE REVIEW

File: pages/agent/v3/elastic_ci_aws.md.erb [hunk 1]
```diff
@@ -364,6 +364,40 @@ You can assess and monitor health and proper function of the Elastic CI Stack us
...
+
+Additionally, on Linux instances only:
+* Docker Daemon logs get sent to the `/buildkite/docker-daemon/{instance_id}` log stream. If docker is having a bad day on your machine, look here.
+* Output from the cloud init process, up until the startup script is initialised, is sent to `/buildkite/cloud-init/output/{instance_id}`. Logs from this group can be useful for inspecting what environment variables were sent to the startup script
+
+On Windows instances only:
+* Logs from the UserData execution process (roughly analogous to the `/buildkite/cloud-init/output` group above) are sent to the `/buildkite/EC2Launch/UserdataExecution/{instance_id}` log stream.
+
+There are a couple of other log groups that the elastic stack sends logs to, but their use cases are pretty specific. For a full accounting of what logs are sent to CloudWatch, see the config for [Linux](https://github.com/buildkite/elastic-ci-stack-for-aws/blob/master/packer/linux/conf/cloudwatch-agent/config.json) and [Windows](witter.com/).
```
Comment (plaindocs) [inline pages/agent/v3/elastic_ci_aws.md.erb:383]: @moskyb got a better link for [Windows](witter.com/) ? ;-)

File: pages/agent/v3/elastic_ci_aws.md.erb [hunk 2]
```diff
@@ -364,6 +364,40 @@ You can assess and monitor health and proper function of the Elastic CI Stack us
 
 ## Troubleshooting
 
+### General Troubleshooting Tips
```
Comment (karensawrey) [inline pages/agent/v3/elastic_ci_aws.md.erb:367]:
```suggestion
### General troubleshooting tips
```

File: pages/agent/v3/elastic_ci_aws.md.erb [hunk 3]
```diff
@@ -364,6 +364,40 @@ You can assess and monitor health and proper function of the Elastic CI Stack us
...
+
+The Elastic CI Stack, as a cloud service, can fail in a variety of interesting ways - too many to provide specific troubleshooting advice here! Luckily, the Elastic CI Stack provides a variety of ways to debug exactly what's going on in failure cases.
+
```
Comment (karensawrey) [inline pages/agent/v3/elastic_ci_aws.md.erb:370]:
```suggestion
The Elastic CI Stack, as a cloud service, can fail in all kinds of interesting ways - too many to provide specific troubleshooting advice for each one! However, the Elastic CI Stack provides a variety of ways to debug the exact failure cases.

```

File: pages/agent/v3/elastic_ci_aws.md.erb [hunk 4]
```diff
@@ -364,6 +364,40 @@ You can assess and monitor health and proper function of the Elastic CI Stack us
...
+#### CloudWatch Logs
+While running, the Elastic CI Stack sends a variety of logs to various CloudWatch log streams:
+* Buildkite Agent logs get sent to the `buildkite/buildkite-agent/{instance_id}` log stream. If there are problems within the agent itself, the agent logs should help diagnose.
+* Output from an Elastic CI Stack instance's startup script ([Linux](https://github.com/buildkite/elastic-ci-stack-for-aws/blob/master/packer/linux/conf/bin/bk-install-elastic-stack.sh) or [Windows](https://github.com/buildkite/elastic-ci-stack-for-aws/blob/master/packer/windows/conf/bin/bk-install-elastic-stack.ps1)) get sent to the `/buildkite/elastic-stack/{instance_id}` log stream. If an instance is failing to launch cleanly, it's often a problem with the startup script, making this log stream especially useful for debugging problems with the elastic stack
+
+Additionally, on Linux instances only:
+* Docker Daemon logs get sent to the `/buildkite/docker-daemon/{instance_id}` log stream. If docker is having a bad day on your machine, look here.
```
Comment (plaindocs) [inline pages/agent/v3/elastic_ci_aws.md.erb:377]: bad day, LOL

File: pages/agent/v3/elastic_ci_aws.md.erb [hunk 5]
```diff
@@ -364,39 +364,32 @@ You can assess and monitor health and proper function of the Elastic CI Stack us
...
-There are a couple of other log groups that the elastic stack sends logs to, but their use cases are pretty specific. For a full accounting of what logs are sent to CloudWatch, see the config for [Linux](https://github.com/buildkite/elastic-ci-stack-for-aws/blob/master/packer/linux/conf/cloudwatch-agent/config.json) and [Windows](witter.com/).
-
-#### Shelling into Elastic Stack instances
-
-Sometimes, looking at the logs isn't quite enough to figure out what's going on in your instances. In these cases, it can be useful to go to the scene of the crime, and tool around in the problematic instance to see what's going on. The Elastic CI Stack provides two distinct ways to do this:
-
-##### Through SSH (if enabled)
 
-If your Elastic CI Stack has been configured to allow SSH access through use of the `AuthorizedUsersUrl` parameter, you can run `ssh <some instance id>` in your terminal, and end up on the instance.
+* Logs from the UserData execution process (analogous to the `/buildkite/cloud-init/output` group above) are sent to the `/buildkite/EC2Launch/UserdataExecution/{instance_id}` log stream.
```
Comment (karensawrey) [inline pages/agent/v3/elastic_ci_aws.md.erb:383]:
```suggestion
* Logs from the UserData execution process (similar to the `/buildkite/cloud-init/output` group above) are sent to the `/buildkite/EC2Launch/UserdataExecution/{instance_id}` log stream.
```

**Review by karensawrey — APPROVED** [review]

File: pages/agent/v3/elastic_ci_aws.md.erb [hunk 6]
```diff
@@ -364,39 +364,32 @@ You can assess and monitor health and proper function of the Elastic CI Stack us
 
 ## Troubleshooting
 
-### General Troubleshooting Tips
+The Elastic CI Stack can be hard to troubleshoot, but here are some ways to debug exactly what's going on, and some solutions for specific situtions.
```
[the remark this answers is not in the document]
Comment (pda) [reply pages/agent/v3/elastic_ci_aws.md.erb]: Maybe something more general (and equally true) like “infrastructure as code can be challenging to troubleshoot”?

[the remark this answers is not in the document]
Comment (plaindocs) [reply pages/agent/v3/elastic_ci_aws.md.erb]: Haha, I didn't see that as dunking, but sure can! Thanks

**Pull request merged**